### PR TITLE
feat: Use `PopupRoot` everywhere

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 0dde4ae</samp>

This pull request refactors the `Popup` class for iOS and Android platforms, adds support for native view anchors, and improves the code quality and safety using C# 9 features and nullable reference types. It also deletes an unused file and makes some minor formatting changes.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Popup/Popup.cs [hunk 1]
```diff
@@ -311,10 +311,10 @@ private Brush GetPanelBackground()
 	/// </summary>
 	internal Brush LightDismissOverlayBackground
 	{
-		get { return (Brush)GetValue(LightDismissOverlayBackgroundProperty); }
-		set { SetValue(LightDismissOverlayBackgroundProperty, value); }
+		get => return (Brush)GetValue(LightDismissOverlayBackgroundProperty);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Popup/Popup.cs:314]:
```suggestion
		get => (Brush)GetValue(LightDismissOverlayBackgroundProperty);
```

**Comment by Youssef1313** [comment]
I think we are not hitting this condition when we should hit it?

https://github.com/unoplatform/uno/blob/a7d93b68dc360107eedbf7039b4fac0f9e2a70f1/src/Uno.UI/UI/Xaml/UIElement.cs#L527

**Review by ajpinedam — APPROVED** [review]
🚀🚀

**Pull request merged**